api/src/services/import-services/critter/critter-header-configs.ts

```diff
@@ -0,0 +1,183 @@
+import { countBy, get } from 'lodash';
+import { z } from 'zod';
+import { CSVConfigUtils } from '../../../utils/csv-utils/csv-config-utils';
+import { CSVError, CSVParams } from '../../../utils/csv-utils/csv-config-validation.interface';
+import { getCSVParamsError, validateZodCell } from '../../../utils/csv-utils/csv-header-configs';
+import { CritterCSVConfig } from './import-critters-service';
+
+/**
+ * Get the critter alias cell validator.
+ *
+ * @param {Set<string>} surveyAliases The survey aliases.
+ * @param {CSVConfigUtils<CritterCSVConfig>} configUtils The CSV config utils.
+ * @returns {*} {(params: CSVParams) => CSVError[]} The validate cell callback
+ */
+export const getCritterAliasCellValidator = (
+  surveyAliases: Set<string>,
+  configUtils: CSVConfigUtils<CritterCSVConfig>
+): ((params: CSVParams) => CSVError[]) => {
+  const rowAliases = configUtils.getCellValues('ALIAS');
+  const rowAliasCounts = countBy(rowAliases.map((alias) => String(alias).toLowerCase()));
+
+  return (params: CSVParams) => {
+    const cellErrors = validateZodCell(params, z.string().trim().min(1).max(50));
+
+    if (cellErrors.length) {
+      return cellErrors;
+    }
+
+    if (surveyAliases.has(String(params.cell))) {
+      cellErrors.push({
+        error: `Critter alias already exists in the Survey`,
+        solution: `Update the alias to be unique`,
+        ...getCSVParamsError(params)
+      });
+    }
+
+    if (rowAliasCounts[String(params.cell).toLowerCase()] > 1) {
+      cellErrors.push({
+        error: `Critter alias already exists in the CSV`,
+        solution: `Update the alias to be unique`,
+        ...getCSVParamsError(params)
+      });
+    }
+
+    return cellErrors;
+  };
+};
+
+/**
+ * Get the critter collection unit cell validator.
+ *
+ * @param {Object} rowDictionary The row dictionary.
+ * @param {CSVConfigUtils<CritterCSVConfig>} configUtils The CSV config utils.
+ * @returns {*} {(params: CSVParams) => CSVError[]} The validate cell callback
+ */
+export const getCritterCollectionUnitCellValidator = (
+  rowDictionary: { [tsn: number]: { [header: string]: { [unit: string]: string } } },
+  configUtils: CSVConfigUtils<CritterCSVConfig>
+): ((params: CSVParams) => CSVError[]) => {
+  return (params: CSVParams) => {
+    const cellErrors = validateZodCell(params, z.string().trim().min(1).max(50).optional());
+
+    if (cellErrors.length || !params.cell) {
+      return cellErrors;
+    }
+
+    const rowTsn = Number(configUtils.getCellValue('ITIS_TSN', params.row));
+    const cellValue = String(params.cell).toLowerCase();
+
+    const rowDictionaryTsn = get(rowDictionary, rowTsn);
+    const rowDictionaryHeader = get(rowDictionary, [rowTsn, params.header]);
+    const rowDictionaryUnit = get(rowDictionary, [rowTsn, params.header, cellValue]);
```

Comment:
minor nit: If `rowDictionaryTsn` is invalid, you won't need to have calculated `rowDictionaryHeader` or `rowDictionaryUnit`.

So you could move those down, after each check, and then just make them all `if` statements with their own returns

Reply:
Similar for the `getCritterSexCellValidator` function